packages/user-mgnt/src/features/system/handler.ts

```diff
@@ -321,6 +328,24 @@ export async function getSystemHandler(
   }
 }
 
+export async function getAllSystemsHandler() {
+  const systems: ISystemModel[] = await System.find()
+  const allSystemsResponseArray: AllSystemResponse[] = []
+
+  systems.forEach((system) => {
+    const allSystemsResponse = {} as AllSystemResponse
+    const systemName = `${system.name[0]?.given || ''} ${
+      system.name[0]?.family || ''
+    }`.trim()
+    allSystemsResponse.client_id = system.client_id
+    allSystemsResponse.name = systemName
+    allSystemsResponse.sha_secret = system.sha_secret
+    allSystemsResponse.status = system.status
+    allSystemsResponseArray.push(allSystemsResponse)
+  })
+  return allSystemsResponseArray
+}
```

Comment:
Would a `.map` be more easy to use here:
```
export async function getAllSystemsHandler() {
  const systems = await System.find()

  return systems.map((system) => {
    const systemName = `${system.name[0]?.given || ''} ${
      system.name[0]?.family || ''
    }`.trim()

    return {
      client_id: system.client_id,
      name: systemName,
      sha_secret: system.sha_secret,
      status: system.status
    }
  })
}
```

Later, when we refactor the system name (it's an upcoming issue), we could even change this to something like:
```
export async function getAllSystemsHandler() {
  return System.find().select('client_id name sha_secret status');
}
```

Reply:
Yeah that can work, looks shorter